precompiles/crowdloan-rewards/src/tests.rs

```diff
@@ -0,0 +1,304 @@
+// Copyright 2019-2021 PureStake Inc.
+// This file is part of Moonbeam.
+
+// Moonbeam is free software: you can redistribute it and/or modify
+// it under the terms of the GNU General Public License as published by
+// the Free Software Foundation, either version 3 of the License, or
+// (at your option) any later version.
+
+// Moonbeam is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+// GNU General Public License for more details.
+
+// You should have received a copy of the GNU General Public License
+// along with Moonbeam.  If not, see <http://www.gnu.org/licenses/>.
+
+use crate::mock::Crowdloan;
+use crate::mock::{
+	events, evm_test_context, precompile_address, roll_to, Call, ExtBuilder, Origin, Precompiles,
+	TestAccount::Alice, TestAccount::Bob, TestAccount::Charlie,
+};
+use crate::PrecompileOutput;
+use frame_support::{assert_ok, dispatch::Dispatchable};
+use pallet_crowdloan_rewards::{Call as CrowdloanCall, Event as CrowdloanEvent};
+use pallet_evm::Call as EvmCall;
+use pallet_evm::{ExitError, ExitSucceed, PrecompileSet};
+use precompiles_utils::solidity_conversions::{bool_to_solidity_bytes, u256_to_solidity_bytes};
+use sp_core::U256;
+
+#[test]
+fn selector_less_than_four_bytes() {
+	ExtBuilder::default().build().execute_with(|| {
+		// This selector is only three bytes long when four are required.
+		let bogus_selector = vec![1u8, 2u8, 3u8];
+
+		// Expected result is an error stating there are too few bytes
+		let expected_result = Some(Err(ExitError::Other(
+			"input length less than 4 bytes".into(),
+		)));
+
+		assert_eq!(
+			Precompiles::execute(
+				precompile_address(),
+				&bogus_selector,
+				None,
+				&evm_test_context(),
+			),
+			expected_result
+		);
+	});
+}
+
+#[test]
+fn no_selector_exists_but_length_is_right() {
+	ExtBuilder::default().build().execute_with(|| {
+		// This selector is only three bytes long when four are required.
+		let bogus_selector = vec![1u8, 2u8, 3u8, 4u8];
+
+		// Expected result is an error stating there are too few bytes
+		let expected_result = Some(Err(ExitError::Other(
+			"No crowdloan rewards wrapper method at given selector".into(),
+		)));
+
+		assert_eq!(
+			Precompiles::execute(
+				precompile_address(),
+				&bogus_selector,
+				None,
+				&evm_test_context(),
+			),
+			expected_result
+		);
+	});
+}
+
+#[test]
+fn is_contributor_returns_false() {
+	ExtBuilder::default()
+		.with_balances(vec![(Alice, 1000)])
+		.build()
+		.execute_with(|| {
+			let selector = hex_literal::hex!("53440c90");
```

Comment:
It'd be cool if these tests could verify the function selector calculation as well, like in #602 [here](https://github.com/PureStake/moonbeam/pull/602#discussion_r670877226)